clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3036,18 +3068,21 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
   if (D->isInvalidDecl())
     return;
 
+  S.CheckDeprecatedSYCLAttributeSpelling(AL);
+
   Expr *XDimExpr = AL.getArgAsExpr(0);
 
-  // If no attribute argument is specified, set to default value '1'
-  // for second and third attribute argument in ReqdWorkGroupSizeAttr
-  // for only with intel::reqd_work_group_size spelling.
+  // If no attribute argument is specified, set the second and third argument
+  // to the default value 1, but only if the sycl:: or intel::
```

Comment:
It turns out that we've read the spec somewhat wrong here. The attribute does not have default arguments and each form with different parameter arity is expected to behave as a distinct attribute (see https://github.com/KhronosGroup/SYCL-Docs/issues/142).

I think a follow-up patch should detangle this situation as it's not really being made worse by these changes.

Reply:
Please open a bug report (to modify the attribute handling) for this unless one already exists. We've had a lot of issues with these set of attributes and I'm worried we're losing information between reviews

Reply:
Github Issues: https://github.com/intel/llvm/issues/3743